File: src/my_cpp_pkg/src/add_two_ints_server.cpp
# include "rclcpp/rclcpp.hpp"
# include "example_interfaces/srv/add_two_ints.hpp"


class AddTwoIntsServer : public rclcpp::Node
{
    public:
        AddTwoIntsServer() : Node("add_two_ints_server")
        {
            server_  = this->create_service<example_interfaces::srv::AddTwoInts>(
                "add_two_ints",
                std::bind(&AddTwoIntsServer::callback_add_two_ints, this,
                std::placeholders::_1, std::placeholders::_2));
            RCLCPP_INFO(this->get_logger(), "Service 'add_two_ints' is ready.");
        }     
    private:
        rclcpp::Service<example_interfaces::srv::AddTwoInts>::SharedPtr server_;

        void callback_add_two_ints(
            const example_interfaces::srv::AddTwoInts::Request::SharedPtr request,
            const example_interfaces::srv::AddTwoInts::Response::SharedPtr response)
        {
            response->sum = request->a + request->b;
            RCLCPP_INFO(this->get_logger(), "Incoming request: a=%d, b=%d", (int)request->a, (int)request->b);
            RCLCPP_INFO(this->get_logger(), "Sending back response: [%d]", (int)response->sum);
        }        
};

int main(int argc, char **argv)
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<AddTwoIntsServer>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}

File: src/my_py_pkg/my_py_pkg/add_two_ints_server.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts

class AddTwoIntsServer(Node):
    def __init__(self):
        super().__init__('add_two_ints_server')
        self.srv = self.create_service(AddTwoInts, 
                                       'add_two_ints', 
                                       self.add_two_ints_callback)
        self.get_logger().info('Add Two Ints Service is ready.')

    def add_two_ints_callback(self, 
                              request: AddTwoInts.Request, 
                              response: AddTwoInts.Response):
        
        response.sum = request.a + request.b
        self.get_logger().info(f'Received request: a={request.a}, b={request.b}, sum={response.sum}')
        return response

def main(args=None):
    rclpy.init(args=args)
    add_two_ints_server = AddTwoIntsServer()
    rclpy.spin(add_two_ints_server)
    add_two_ints_server.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()